subprojects/gradle/build-trace/src/main/kotlin/com/avito/android/build_trace/internal/TaskExt.kt

```diff
@@ -0,0 +1,29 @@
+package com.avito.android.build_trace.internal
+
+import org.gradle.api.Task
+import org.gradle.api.internal.TaskInternal
+import org.gradle.api.tasks.TaskDependency
+
+internal val Task.type: Class<out Task>
+    get() = (this as TaskInternal).taskIdentity.type
+
+internal val Task.predecessors: Set<Task>
+    get() {
+        val dependenciesByInputs = project.gradle.taskGraph.getDependencies(this)
+        return (
+            dependenciesByInputs +
+                taskDependencies.safeDependencies(this) +
+                mustRunAfter.safeDependencies(this) +
+                shouldRunAfter.safeDependencies(this)
+            )
+            .toSet()
+    }
+
+internal fun TaskDependency.safeDependencies(task: Task): Set<Task> {
```

Comment:
Where do we use this Set?

Reply:
In this file, for example. 
To resolve dependent tasks described as `dependsOn` / `mustRunAfter` / `shouldRunAfter` / `finalizedBy`

Reply:
So it's incorrect to return emptySet

Reply:
In general - incorrect
For https://issuetracker.google.com/issues/142268503 we have to preserve at least most of the data.
I'll change the logic:
https://github.com/avito-tech/avito-android/pull/973/files/2dbe5c6cd2cc7f4ceb0d37655aaa4b7200902a6b#r631650972

Reply:
Removed the check at all because couldn't reproduce the issue.
https://github.com/avito-tech/avito-android/pull/985/commits/649bc7548b8c45d8cfbd158643d2681249186ab6